llama_stack/providers/inline/vector_io/faiss/faiss.py

```diff
@@ -98,7 +98,9 @@ async def add_chunks(self, chunks: List[Chunk], embeddings: NDArray):
         # Save updated index
         await self._save_index()
 
-    async def query(self, embedding: NDArray, k: int, score_threshold: float) -> QueryChunksResponse:
+    async def query(
+        self, embedding: NDArray, k: int, score_threshold: float, query_str: None, search_mode: None
```

Comment:
can we instead introduce a new type of provider like "keyword_search_io"? Presumably there can be other full text search providers like elasticsearch or postgres ts_query that can be added in addition to sqlite-fts5.

Reply:
@raghotham maybe we should discuss this in an RFC?

I personally think that vector, keyword, and standard primary key lookup are all retrieval strategies for a database. I would worry making three separate providers may introduce too much of an abstraction.

That is largely how we've implemented things in @feast-dev (the sqlite example: https://github.com/feast-dev/feast/pull/5082), which I had pointed @varshaprasad96 to prior to doing the implementation. Of course, that may not be how we want to design these things here, so definitely open to your thoughts here. I just wanted to be transparent with how I saw things.